src/modules/move/hooks/useCancelable.jsx

```diff
@@ -0,0 +1,29 @@
+import { useEffect, useState } from 'react'
+
+import { cancelable } from 'cozy-client'
+
+const useCancelable = () => {
+  const [promises, setPromises] = useState([])
+
+  useEffect(() => {
+    // unregister cancelables when component will unmount
+    return () => {
+      promises.forEach(p => p.cancel())
+      setPromises([])
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  const registerCancelable = promise => {
+    if (!promises) setPromises([])
+    const cancelableP = cancelable(promise)
+    setPromises([...promises, cancelableP])
+    return cancelableP
+  }
+
+  return {
+    registerCancelable
+  }
+}
+
+export { useCancelable }
```

Comment:
Suggestion: Refactor useCancelable to use a ref for storing promises instead of state to prevent unnecessary re-renders. Also, memoize the registerCancelable function using useCallback for better performance.

```jsx
const useCancelable = () => {
  const promisesRef = useRef([]);

  useEffect(() => {
    // Cleanup function to cancel all promises
    return () => {
      promisesRef.current.forEach(p => p.cancel());
      promisesRef.current = [];
    };
  }, []);

  const registerCancelable = useCallback(promise => {
    const cancelableP = cancelable(promise);
    promisesRef.current.push(cancelableP);
    return cancelableP;
  }, []);

  return {
    registerCancelable
  };
};
```

Reply:
I had just extracted the function without challenging it. Nice idea for the optimisation